File: row_major.c
#include <stdlib.h>
#include <stdio.h>
#include <stdint.h>

#define MATRIX_SIZE 10000

int main(void) {
    int **matrix = malloc(MATRIX_SIZE * sizeof(*matrix));
    for (int i = 0; i < MATRIX_SIZE; i++) {
        matrix[i] = malloc(MATRIX_SIZE * sizeof(**matrix));
        for (int j = 0; j < MATRIX_SIZE; j++) {
            matrix[i][j] = i + j;
        }
    }

    uint64_t sum = 0;
    for (int i = 0; i < MATRIX_SIZE; i++) {
        for (int j = 0; j < MATRIX_SIZE; j++) {
            sum += matrix[i][j];
        }
    }

    printf("%llu\n", sum);

    for (int i = 0; i < MATRIX_SIZE; i++) {
        free(matrix[i]);
    }
    free(matrix);

    return 0;
}

File: row_major.rs
const MATRIX_SIZE: usize = 10000;

fn main() {
    let mut matrix = vec![vec![0i32; MATRIX_SIZE]; MATRIX_SIZE];

    for i in 0..MATRIX_SIZE {
        for j in 0..MATRIX_SIZE {
            matrix[i][j] = (i + j) as i32;
        }
    }

    let mut sum: u64 = 0;
    for i in 0..MATRIX_SIZE {
        for j in 0..MATRIX_SIZE {
            sum += matrix[i][j] as u64;
        }
    }

    println!("{}", sum);
}
